From 437ae9829c23224d8bb288f5dce818167b0e0d1a Mon Sep 17 00:00:00 2001
From: Alex Marginean <alexandru.marginean@nxp.com>
Date: Tue, 24 Apr 2018 14:38:54 +0300
Subject: [PATCH 137/706] enetc: set SI based VLAN bits per BG 123

These are required for VF VLAN.

Signed-off-by: Alex Marginean <alexandru.marginean@nxp.com>
(cherry picked from commit 51f847cdf8e9937b267e4d4cbb00be1c6268068f)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/freescale/enetc/enetc_hw.h | 2 ++
 drivers/net/ethernet/freescale/enetc/enetc_pf.c | 1 +
 2 files changed, 3 insertions(+)

diff --git a/drivers/net/ethernet/freescale/enetc/enetc_hw.h b/drivers/net/ethernet/freescale/enetc/enetc_hw.h
index 4a87f22a7fd4..ac6a55747d67 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc_hw.h
+++ b/drivers/net/ethernet/freescale/enetc/enetc_hw.h
@@ -190,6 +190,8 @@ enum enetc_bdr_type {TX, RX};
 #define ENETC_PSICFGR0(n)	(0x00940 + (n) * 0xc)  /* n = SI index */
 #define ENETC_PSICFGR0_SET_TXBDR(val)	((val) & 0xff)
 #define ENETC_PSICFGR0_SET_RXBDR(val)	(((val) & 0xff) << 16)
+#define ENETC_PSICFGR0_VTE	BIT(12)
+#define ENETC_PSICFGR0_SIVIE	BIT(14)
 #define ENETC_PSICFGR0_ASE	BIT(15)
 #define ENETC_PSICFGR0_SIVC(bmp)	(((bmp) & 0xff) << 24) /* VLAN_TYPE */
 
diff --git a/drivers/net/ethernet/freescale/enetc/enetc_pf.c b/drivers/net/ethernet/freescale/enetc/enetc_pf.c
index 21817a94339d..a2e371635fde 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc_pf.c
+++ b/drivers/net/ethernet/freescale/enetc/enetc_pf.c
@@ -478,6 +478,7 @@ static void enetc_port_si_configure(struct enetc_si *si)
 
 	/* Configure the SIs for each available VF */
 	val = ENETC_PSICFGR0_SIVC(ENETC_VLAN_TYPE_C | ENETC_VLAN_TYPE_S);
+	val |= ENETC_PSICFGR0_VTE | ENETC_PSICFGR0_SIVIE;
 
 	if (num_rings) {
 		num_rings /= pf->total_vfs;
-- 
2.17.1

